win32ss/user/user32/windows/dialog.c

```diff
@@ -2635,12 +2635,14 @@ IsDialogMessageW(
                           break;
                   }
 
-                 if (hwndNext && SendMessageW( hwndNext, WM_GETDLGCODE, lpMsg->wParam, (LPARAM)lpMsg ) == (DLGC_BUTTON | DLGC_RADIOBUTTON))
+                 if (hwndNext &&
+                     ((SendMessageW(hwndNext, WM_GETDLGCODE, lpMsg->wParam, (LPARAM)lpMsg) &
+                       (DLGC_BUTTON | DLGC_RADIOBUTTON)) == (DLGC_BUTTON | DLGC_RADIOBUTTON)))
                  {
                      SetFocus( hwndNext );
                      if ((GetWindowLongW( hwndNext, GWL_STYLE ) & BS_TYPEMASK) == BS_AUTORADIOBUTTON &&
                          SendMessageW( hwndNext, BM_GETCHECK, 0, 0 ) != BST_CHECKED)
-                         SendMessageW( hwndNext, BM_CLICK, 1, 0 );
+                         SendMessageW(hwndNext, BM_CLICK, 0, 0);
```

Comment:
Would the following work and be more consistent?
```suggestion
                         SendMessageW(hwndNext, BM_SETCHECK, BST_CHECKED, 0);
```

Reply:
Can `BN_CLICKED` be generated by that?

Reply:
> Can `BN_CLICKED` be generated by that?

I would guess it cannot. My question was more like 'is that needed'? Anyway.

--

> `BM_CLICK`'s `wParam` must be zero.

@katahiromz, Are the followings affected too?
https://github.com/reactos/reactos/blob/d55add359cc96b38ce4e9ebf4eee845dfd64ba69/base/applications/charmap/settings.c#L96
https://github.com/reactos/reactos/blob/d55add359cc96b38ce4e9ebf4eee845dfd64ba69/modules/rostests/winetests/user32/msg.c#L6834

Reply:
```c
SendDlgItemMessage(hCharmapDlg, IDC_CHECK_ADVANCED, BM_CLICK, MF_CHECKED, 0);
```
Fixed.

```c
{ BM_CLICK, sent|wparam|lparam, 1, 0 }, 
```
I'm not sure.